presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -587,6 +595,112 @@ private void testPredicating(Session session, FileFormat fileFormat)
         dropTable(session, "test_predicating_on_real");
     }
 
+    @Test
+    public void testMetadataDeleteSimple()
+    {
+        testWithAllFileFormats(this::testMetadataDeleteSimpleForFormat);
+    }
+
+    private void testMetadataDeleteSimpleForFormat(Session session, FileFormat format)
+    {
+        assertUpdate(session, format("CREATE TABLE test_metadata_delete_simple (col1 BIGINT, col2 BIGINT) WITH (format = '%s', partitioning = ARRAY['col1'])", format.name()));
+        assertUpdate(session, "INSERT INTO test_metadata_delete_simple VALUES(1, 100), (1, 101), (1, 102), (2, 200), (2, 201), (3, 300)", 6);
+        assertQueryFails(
+                session,
+                "DELETE FROM test_metadata_delete_simple WHERE col1 = 1 AND col2 > 101",
+                "This connector only supports delete where one or more partitions are deleted entirely");
+        assertQuery(session, "SELECT COUNT(*) FROM test_metadata_delete_simple", "SELECT 6");
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete_simple$partitions\"", "SELECT 3");
+        assertUpdate(session, "DELETE FROM test_metadata_delete_simple WHERE col1 = 1");
+        assertQuery(session, "SELECT COUNT(*) FROM test_metadata_delete_simple", "SELECT 3");
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete_simple$partitions\"", "SELECT 2");
+        dropTable(session, "test_metadata_delete_simple");
+    }
+
+    @Test
+    public void testMetadataDelete()
+    {
+        testWithAllFileFormats(this::testMetadataDeleteForFormat);
+    }
+
+    private void testMetadataDeleteForFormat(Session session, FileFormat format)
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_metadata_delete " +
+                "(" +
+                "  ORDER_KEY BIGINT," +
+                "  LINE_NUMBER INTEGER," +
+                "  LINE_STATUS VARCHAR" +
+                ") " +
+                "WITH (" +
+                format(" format = '%s', partitioning = ARRAY[ 'LINE_NUMBER', 'LINE_STATUS' ]", format.name()) +
+                ") ";
+
+        assertUpdate(session, createTable);
+
+        assertUpdate(session, "" +
+                        "INSERT INTO test_metadata_delete " +
+                        "SELECT orderkey, linenumber, linestatus " +
+                        "FROM tpch.tiny.lineitem",
+                "SELECT count(*) FROM lineitem");
+
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete$partitions\"", "SELECT 14");
+
+        assertUpdate(session, "DELETE FROM test_metadata_delete WHERE LINE_STATUS='F' AND LINE_NUMBER=CAST(3 AS INTEGER)");
```

Comment:
This cast shouldn't be necessary as the `3` literal should be `integer`.

Nit: lowercase column names and add space around operators (same for other queries)

Reply:
Cast removed and column names and whitespace fixed everywhere.